homeassistant/components/todoist/calendar.py

```diff
@@ -122,6 +122,11 @@ async def async_setup_platform(
     coordinator = TodoistCoordinator(hass, _LOGGER, SCAN_INTERVAL, api)
     await coordinator.async_config_entry_first_refresh()
```

Comment:
Note: this line feels wrong to me.
I think `async_config_entry_first_refresh` shouldn't be used in this context.
But that is a bug that you be addressed maybe by the code owner in a follow-up PR.

Reply:
Any particular reason why?  I changed it from `coordinator.async_refresh` after seeing this comment: https://github.com/boralyl/home-assistant/blob/48fca3bb2727b393d29f2246bc4df98ba8c35ed6/homeassistant/helpers/update_coordinator.py#L76-L80

Reply:
`async_config_entry_first_refresh` raises `ConfigEntryNotReady`, and should only be used within a config-entry setup.
In your case, you should be raising `PlatformNotReady` if it fails.

Reply:
Thanks!